docs/DeveloperGuide.md

```diff
@@ -1853,44 +1853,698 @@ testers are expected to do more _exploratory_ testing.
+
+   4. Test Case: `delivery view -1`{.swift}.</br>
+      Expected: No new Delivery details are shown. Error indicating invalid command format is
+      shown in the feedback message.
+
+   5. Test Case: `delivery view a`{.swift}.</br>
+      Expected: Similar to previous.
+
+   6. Test Case: `delivery view 2`{.swift}.</br>
+      Expected: No new Delivery details are shown. Error indicating invalid delivery ID is
+      shown in the feedback message.
+
+### List Deliveries
+
+1. List the Deliveries stored in the application.
+
+   1. Prerequisites: Logged-in into the application with the `login`{.swift} command.
+      There are two Customers with an ID of 1 and 2 stored in the application.
+      There are two deliveries stored in the application. 
+      The first with ID 1, Customer ID of 1, delivery status `CREATED`{.swift}, expected delivery date `2023-12-03`{.swift}.
+      The second with ID 2, Customer ID of 2, delivery status `SHIPPED`{.swift}, expected delivery date `2023-12-04`{.swift}.
+      The current date is `2023-12-03`{.swift}.
+
+   2. Test Case: `delivery list`{.swift}.</br>
+      Expected: All Deliveries are listed sorted in descending expected delivery date. 
+      A message indicating that deliveries have been listen is shown in the feedback message.
+
+   3. Test Case: `delivery list --status CREATED`{.swift}.</br>
+      Expected: The Delivery with an ID of 1 is listed. A message indicating that Deliveries have been listed is
+      shown in the feedback message.
+
+   4. Test Case: `delivery list --customer 1`{.swift}.</br>
+      Expected: Similar to previous.
+
+   5. Test Case: `delivery list --date today`{.swift}.</br>
+      Expected: Similar to previous.
+
+   6. Test Case: `delivery list --date 2023-12-04`{.swift}.</br>
+      Expected: The Delivery with an ID of 2 is listed. A message indicating that Deliveries have been listed is
+      shown in the feedback message.
+
+   7. Test Case: `delivery list --sort ASC`{.swift}.</br>
+      Expected: All Deliveries are listed sorted in ascending expected delivery date.
+      A message indicating that deliveries have been listen is shown in the feedback message.
+
+   8. Test Case: `delivery list --status INVALID`{.swift}.</br>
+      Expected: No Deliveries are listed. An Error indicating delivery status constraints
+      is shown in the feedback message.
+
+   9. Test Case: `delivery list --customer 0`{.swift}.</br>
+      Expected: No Deliveries are listed. An Error indicating Customer ID constraints
+      is shown in the feedback message.
+
+   10. Test Case: `delivery list --date 2023-13-04`{.swift}.</br>
+      Expected: No Deliveries are listed. An Error indicating expected delivery date constraints
+      is shown in the feedback message.
+
+   11. Test Case: `delivery list --sort random`{.swift}.</br>
+       Expected: No Deliveries are listed. An Error indicating sort constraints
+       is shown in the feedback message.
+
```

Comment:
Should we test for combination of `delivery list`. For example `delivery list --sort asc --date today --status CREATED`